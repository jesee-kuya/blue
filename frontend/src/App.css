.app {
  display: flex;
  flex-direction: column;
  height: 100vh;
  max-width: 1200px;
  margin: 0 auto;
  background: #f5f5f5;
}

.app-header {
  background: #2563eb;
  color: white;
  padding: 1rem 2rem;
  box-shadow: 0 2px 4px rgba(0,0,0,0.1);
}

.app-header h1 {
  margin: 0;
  font-size: 1.5rem;
  font-weight: 600;
}

.error-banner {
  background: #fee2e2;
  color: #dc2626;
  padding: 0.75rem 1rem;
  margin-top: 1rem;
  border-radius: 0.5rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.error-banner button {
  background: none;
  border: none;
  color: #dc2626;
  font-size: 1.25rem;
  cursor: pointer;
  padding: 0;
  width: 24px;
  height: 24px;
}

.chat-container {
  display: flex;
  flex-direction: column;
  flex: 1;
  overflow: hidden;
}

.input-section {
  background: white;
  border-top: 1px solid #e5e7eb;
  padding: 1rem;
}

@media (max-width: 768px) {
  .app-header {
    padding: 1rem;
  }
  
  .app-header h1 {
    font-size: 1.25rem;
  }
  
  .input-section {
    padding: 0.75rem;
  }
}
