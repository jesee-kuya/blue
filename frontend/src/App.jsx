import { useState, useEffect } from 'react';
import MessageList from './components/MessageList';
import MessageInput from './components/MessageInput';
import FileUploader from './components/FileUploader';
import ErrorBoundary from './components/ErrorBoundary';
import { sendMessage, getMarketingCopy } from './services/api';
import './App.css';

function App() {
  const [messages, setMessages] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [attachments, setAttachments] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    // Add welcome message
    setMessages([{
      id: Date.now(),
      sender: 'assistant',
      text: 'Hello! I\'m your Blue marketplace assistant. I can help you search for products, generate marketing copy, and more. How can I assist you today?',
      timestamp: Date.now(),
      type: 'text'
    }]);
  }, []);

  const addMessage = (sender, text, type = 'text', data = null) => {
    const message = {
      id: Date.now() + Math.random(),
      sender,
      text,
      timestamp: Date.now(),
      type,
      data
    };
    setMessages(prev => [...prev, message]);
    return message;
  };

  const handleSendMessage = async (messageText, messageAttachments) => {
    setError(null);
    
    // Add user message
    addMessage('user', messageText || 'Sent attachments');
    
    setIsLoading(true);
    
    try {
      // Determine if this is a marketing request
      const isMarketingRequest = messageText.toLowerCase().includes('marketing') || 
                                messageText.toLowerCase().includes('ad copy') ||
                                messageText.toLowerCase().includes('campaign');
      
      let response;
      if (isMarketingRequest) {
        response = await getMarketingCopy(messageText);
      } else {
        response = await sendMessage(messageText, messageAttachments);
      }
      
      // Handle different response types
      if (response.search_results && response.search_results.length > 0) {
        addMessage('assistant', 'Here are the products I found:', 'search_results', {
          products: response.search_results
        });
      } else if (response.marketing_copy) {
        addMessage('assistant', 'Here\'s the marketing copy I generated:', 'marketing_copy', {
          marketing: response.marketing_copy
        });
      } else if (response.message) {
        addMessage('assistant', response.message);
      } else {
        addMessage('assistant', 'I received your request but couldn\'t process it properly. Please try again.');
      }
      
    } catch (error) {
      console.error('API Error:', error);
      setError(error.message);
      addMessage('assistant', `Sorry, I encountered an error: ${error.message}`);
    } finally {
      setIsLoading(false);
      setAttachments([]);
    }
  };

  const handleAddAttachment = (attachment) => {
    setAttachments(prev => [...prev, attachment]);
  };

  const handleRemoveAttachment = (index) => {
    setAttachments(prev => prev.filter((_, i) => i !== index));
  };

  return (
    <ErrorBoundary>
      <div className="app">
        <header className="app-header">
          <h1>Blue Marketplace Assistant</h1>
          {error && (
            <div className="error-banner" role="alert">
              {error}
              <button onClick={() => setError(null)} aria-label="Dismiss error">Ã—</button>
            </div>
          )}
        </header>
        
        <main className="chat-container">
          <MessageList messages={messages} isLoading={isLoading} />
          
          <div className="input-section">
            <FileUploader 
              onAddAttachment={handleAddAttachment}
              disabled={isLoading}
            />
            <MessageInput
              onSendMessage={handleSendMessage}
              disabled={isLoading}
              attachments={attachments}
              onRemoveAttachment={handleRemoveAttachment}
            />
          </div>
        </main>
      </div>
    </ErrorBoundary>
  );
}

export default App;
