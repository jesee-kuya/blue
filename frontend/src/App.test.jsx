import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('App Component', () => {
  test('renders without crashing', () => {
    render(<App />);
    expect(screen.getByText('Vite + React')).toBeInTheDocument();
  });

  test('displays initial count of 0', () => {
    render(<App />);
    expect(screen.getByText('count is 0')).toBeInTheDocument();
  });

  test('increments count when button is clicked', async () => {
    const user = userEvent.setup();
    render(<App />);

    const button = screen.getByRole('button', { name: /count is 0/i });
    await user.click(button);

    expect(screen.getByText('count is 1')).toBeInTheDocument();
  });

  test('increments count multiple times', async () => {
    const user = userEvent.setup();
    render(<App />);

    const button = screen.getByRole('button', { name: /count is/i });

    await user.click(button);
    await user.click(button);
    await user.click(button);

    expect(screen.getByText('count is 3')).toBeInTheDocument();
  });

  test('renders Vite and React logos with correct links', () => {
    render(<App />);

    const viteLink = screen.getByRole('link', { name: /vite logo/i });
    const reactLink = screen.getByRole('link', { name: /react logo/i });

    expect(viteLink).toHaveAttribute('href', 'https://vite.dev');
    expect(reactLink).toHaveAttribute('href', 'https://react.dev');
  });

  // âœ… FIXED: supports text split by <code> tag
  test('displays HMR instruction text', () => {
    render(<App />);
    const code = screen.getByText('src/App.jsx');
    expect(code).toBeInTheDocument();

    const paragraph = code.closest('p');
    expect(paragraph).toHaveTextContent(/Edit\s+src\/App\.jsx\s+and save to test HMR/);
  });


  test('displays read the docs text', () => {
    render(<App />);
    expect(screen.getByText('Click on the Vite and React logos to learn more')).toBeInTheDocument();
  });

  test('button has correct initial text and updates on click', async () => {
    const user = userEvent.setup();
    render(<App />);

    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('count is 0');

    await user.click(button);
    expect(button).toHaveTextContent('count is 1');
  });

  test('matches snapshot', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot after state change', async () => {
    const user = userEvent.setup();
    const { container } = render(<App />);

    const button = screen.getByRole('button');
    await user.click(button);

    expect(container.firstChild).toMatchSnapshot();
  });
});
