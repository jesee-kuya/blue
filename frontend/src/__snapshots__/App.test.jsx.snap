// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <a
    href="https://vite.dev"
    target="_blank"
  >
    <img
      alt="Vite logo"
      class="logo"
      src="test-file-stub"
    />
  </a>
  <a
    href="https://react.dev"
    target="_blank"
  >
    <img
      alt="React logo"
      class="logo react"
      src="test-file-stub"
    />
  </a>
</div>
`;

exports[`App Component matches snapshot after state change 1`] = `
<div>
  <a
    href="https://vite.dev"
    target="_blank"
  >
    <img
      alt="Vite logo"
      class="logo"
      src="test-file-stub"
    />
  </a>
  <a
    href="https://react.dev"
    target="_blank"
  >
    <img
      alt="React logo"
      class="logo react"
      src="test-file-stub"
    />
  </a>
</div>
`;
